pkg/providers/cloudstack/cloudstack.go

```diff
@@ -769,7 +779,7 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, datacenterConfigSpec v1alpha1
 	}
 
 	if clusterSpec.Cluster.Spec.ProxyConfiguration != nil {
-		fillProxyConfigurations(values, clusterSpec, datacenterConfigSpec)
+		fillProxyConfigurations(values, clusterSpec, clusterSpec.CloudStackDatacenter.Spec)
```

Comment:
Same, remove third arg.
```suggestion
		fillProxyConfigurations(values, clusterSpec)
```